WeatherToday/Sources/Data/NetworkAPI/Core/RequestConfiguration.swift

```diff
@@ -2,38 +2,55 @@
 //  RequestConfiguration.swift
 //
 
-import Alamofire
 import Foundation
+import Moya
 
-protocol RequestConfiguration {
-
-    var baseURL: String { get }
-
-    var endpoint: String { get }
-
-    var method: HTTPMethod { get }
-
-    var url: URLConvertible { get }
-
-    var parameters: Parameters? { get }
+enum RequestConfiguration {
+    case weather(cityName: String)
+}
 
-    var encoding: ParameterEncoding { get }
+extension RequestConfiguration: TargetType {
 
-    var headers: HTTPHeaders? { get }
+    var baseURL: URL { URL(string: Configuration.baseWeatherURL)! }
 
-    var interceptor: RequestInterceptor? { get }
-}
+    var path: String {
+        switch self {
+        case .weather:
+            return "data/2.5/weather"
+        }
+    }
 
-extension RequestConfiguration {
+    var method: Moya.Method {
+        switch self {
+        case .weather:
+            return .get
+        }
+    }
 
-    var url: URLConvertible {
-        let url = URL(string: baseURL)?.appendingPathComponent(endpoint)
-        return url?.absoluteString ?? "\(baseURL)\(endpoint)"
+    var task: Moya.Task {
+
+        switch self {
+        case .weather(let cityName):
+                return Task.requestParameters(
+                    parameters: [
+                        "q": cityName,
+                        "appid": Configuration.weatherApiKey,
+                        "units": "metric"
+                    ],
+                    encoding: URLEncoding.default
+                )
+        }
     }
 
-    var parameters: Parameters? { nil }
+    var headers: [String : String]? { ["Content-Type": "application/json"] }
 
-    var headers: HTTPHeaders? { nil }
+    // Optional Stub
```

Comment:
Please provide more descriptive comment, what is `Optional Stub`, do we have to do something about it in the future: `TODO:`.
I don't think we have to include `sampleData` in this declaration, correct me if I'm wrong.

Reply:
you are right.. this `sampleData` is for creating mock response, will add details comment.

Reply:
Same with comment with documentation.


According to the [source code   ](https://github.com/Moya/Moya/blob/master/Sources/Moya/TargetType.swift), this property is optional, so we don't have to declare it when using the default value.